server/repository/src/db_diesel/temperature_chart/temperature_chart_row.rs

```diff
@@ -1,220 +1,391 @@
+        where
+            $column_name: AppearsOnTable<Join<Left, Right, Inner>>,
+            Left: AppearsInFromClause<super::table> + QuerySource,
+            Right: AppearsInFromClause<super::table> + QuerySource,
+            (Left::Count, Right::Count): Pick<Left, Right>,
+            Self:
+                SelectableExpression<<(Left::Count, Right::Count) as Pick<Left, Right>>::Selection>,
         {
-            out.push_sql(union_prefix);
-            // Only add UNION after first select
-            union_prefix = " UNION ";
+        }
 
-            out.push_sql(" SELECT ");
-            out.push_bind_param::<Timestamp, _>(&from_date)?;
-            out.push_sql(" as from_datetime, ");
-            out.push_bind_param::<Timestamp, _>(&to_date)?;
-            out.push_sql(" as to_datetime, ");
-            out.push_bind_param::<Text, _>(&interval_id)?;
-            out.push_sql(" as interval_id ");
+        impl<Join, On> SelectableExpression<JoinOn<Join, On>> for $column_name where
+            $column_name: SelectableExpression<Join> + AppearsOnTable<JoinOn<Join, On>>
+        {
         }
 
-        out.push_sql(
-            r#" 
+        impl<From> SelectableExpression<SelectStatement<FromClause<From>>> for $column_name
+        where
+            From: QuerySource,
+            $column_name:
+                SelectableExpression<From> + AppearsOnTable<SelectStatement<FromClause<From>>>,
+        {
+        }
+
+        impl<__GB> ValidGrouping<__GB> for $column_name
+        where
+            __GB: IsContainedInGroupBy<$column_name, Output = is_contained_in_group_by::Yes>,
+        {
+            type IsAggregate = is_aggregate::Yes;
+        }
+        impl ValidGrouping<()> for $column_name {
+            type IsAggregate = is_aggregate::No;
+        }
+        impl IsContainedInGroupBy<$column_name> for $column_name {
+            type Output = is_contained_in_group_by::Yes;
+        }
+        impl Column for $column_name {
+            type Table = super::table;
+            const NAME: &'static str = $column_sql;
+        }
+        impl<T> diesel::EqAll<T> for $column_name
+        where
+            T: AsExpression<$column_type>,
+            diesel::dsl::Eq<$column_name, T::Expression>: diesel::Expression<SqlType = Bool>,
+        {
+            type Output = diesel::dsl::Eq<Self, T::Expression>;
+            fn eq_all(self, __diesel_internal_rhs: T) -> Self::Output {
+                use diesel::expression_methods::ExpressionMethods;
+                self.eq(__diesel_internal_rhs)
+            }
+        }
+
+        #[cfg(feature = "postgres")]
+        impl AppearsInFromClause<diesel::query_builder::Only<super::table>> for $column_name {
+            type Count = Once;
+        }
+        #[cfg(feature = "postgres")]
+        impl SelectableExpression<diesel::query_builder::Only<super::table>> for $column_name {}
+    };
+}
+
+#[allow(unused_imports, dead_code, unreachable_pub)]
+pub mod temperature_chart {
```

Comment:
Didn't figure out how to get the previous version working. But really, I don't think there fundamental differences. I started from the generated macro and worked backwards as described in the existing readme.